pkg/sdn/plugin/ovscontroller_test.go

```diff
@@ -0,0 +1,214 @@
+package plugin
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	osapi "github.com/openshift/origin/pkg/sdn/api"
+	"github.com/openshift/origin/pkg/util/ovs"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapiunversioned "k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+func setup(t *testing.T) (ovs.Interface, *ovsController, []string) {
+	ovsif := ovs.NewFake(BR)
+	oc := NewOVSController(ovsif)
+	err := oc.SetupOVS("10.128.0.0/14", "172.30.0.0/16", "10.128.0.0/23", "10.128.0.1", "note:00.00")
+	if err != nil {
+		t.Fatalf("Unexpected error setting up OVS: %v", err)
+	}
+
+	origFlows, err := ovsif.DumpFlows()
+	if err != nil {
+		t.Fatalf("Unexpected error dumping flows: %v", err)
+	}
+
+	return ovsif, oc, origFlows
+}
+
+type flowChangeKind string
+
+const (
+	flowAdded   flowChangeKind = "added"
+	flowRemoved flowChangeKind = "removed"
+)
+
+type flowChange struct {
+	kind    flowChangeKind
+	match   []string
+	noMatch []string
+}
+
+// assertFlowChanges asserts that origFlows and newFlows differ in the ways described by
+// changes, which consists of a series of flows that have been removed from origFlows or
+// added to newFlows. There must be exactly 1 matching flow that contains all of the
+// strings in match and none of the strings in noMatch.
+func assertFlowChanges(origFlows, newFlows []string, changes ...flowChange) error {
+	// copy to avoid modifying originals
+	dup := make([]string, 0, len(origFlows))
+	origFlows = append(dup, origFlows...)
+	dup = make([]string, 0, len(newFlows))
+	newFlows = append(dup, newFlows...)
```

Comment:
it seems like that would take one more line of code to do the same thing though:

    dup := make([]string, len(origFlows))
    copy(dup, origFlows)
    origFlows = dup